src/types/datetime.rs

```diff
@@ -354,6 +368,20 @@ impl PyTimeAccess for PyDateTime {
     }
 }
 
+impl PyTzInfoAccess for PyDateTime {
+    fn get_tzinfo(&self) -> Option<&PyTzInfo> {
+        let tzinfo: &PyTzInfo = unsafe {
+            self.py()
+                .from_borrowed_ptr(PyDateTime_DATE_GET_TZINFO(self.as_ptr()))
+        };
+        if tzinfo.is_none() {
+            None
+        } else {
+            Some(tzinfo)
+        }
```

Comment:
Maybe to avoid the dance with Python `None` we should just read from the FFI structs directly?

```suggestion
        let ptr = self.as_ptr() as *mut ffi::PyDateTime_DateTime;
        unsafe { 
            if (*ptr).has_tzinfo {
                Some(self.py().from_borrowed_ptr((*ptr).tzinfo)
            } else {
                None
            }
        }
```

and similar for the time implementation